### PR TITLE
fix(side-panel): Update the iteration tree list

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Update the iteration tree list when an iteration has been added or edited.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2125-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2125-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2125-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2125-fabric8-planner.badger.fabric8.io

File: src/app/components/iterations-panel/iterations-panel.component.ts [hunk 1]
```diff
@@ -227,6 +227,21 @@ export class IterationComponent implements OnInit, OnDestroy, OnChanges {
       this.allIterations[index] = iteration;
     } else {
       this.allIterations.splice(this.allIterations.length, 0, iteration);
+      //Check if the new iteration has a parent
+      if (!this.iterationService.isTopLevelIteration(iteration)) {
+        let parent = this.iterationService.getDirectParent(iteration, this.allIterations);
+        let parentIndex = this.allIterations.findIndex(i => i.id === parent.id)
```
Comment (vikram-raj) [inline src/app/components/iterations-panel/iterations-panel.component.ts]: semicolon missing.

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2125-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2125-fabric8-planner.badger.fabric8.io

**Pull request merged**